crates/crypto/merkle/src/lib.rs

```diff
@@ -118,15 +111,12 @@ mod tests {
             B256::from_slice(&[0xDD; 32]),
         ];
 
-        let depth = (leaves.len() as f64).log2().floor() as u64;
+        let depth = (leaves.len() as f64).log2().ceil() as u64;
 
-        let node_2: B256 =
-            ethereum_hashing::hash32_concat(leaves[0].as_slice(), leaves[1].as_slice()).into();
-        let node_3: B256 =
-            ethereum_hashing::hash32_concat(leaves[2].as_slice(), leaves[3].as_slice()).into();
+        let node_2: B256 = hash_concat(leaves[0].as_slice(), leaves[1].as_slice());
```

Comment:
```suggestion
        let node_2 = hash_concat(leaves[0].as_slice(), leaves[1].as_slice());
```

same with below we no longer need these types declarations